contrib/snappy-cmake/CMakeLists.txt

```diff
@@ -24,6 +24,12 @@ else ()
     set (SNAPPY_HAVE_SSSE3 0)
 endif ()
 
+if (ARCH_AMD64 AND (ENABLE_AVX OR ENABLE_AVX2 OR ENABLE_AVX512))
```

Comment:
`crc32` is part of `sse4.2`, so we can relax the condition

Reply:
Can it be changed to the following?
```cmake
if (ARCH_AMD64 AND (ENABLE_SSSE3 OR ENABLE_AVX OR ENABLE_AVX2 OR ENABLE_AVX512))
    set (SNAPPY_HAVE_SSSE3 1)
    set (SNAPPY_HAVE_X86_CRC32 1)
else ()
    set (SNAPPY_HAVE_SSSE3 0)
    set (SNAPPY_HAVE_X86_CRC32 0)
endif ()
```
Thanks.

Reply:
currently we have a pretty strange situation. we already set the flag for `ssse3`

https://github.com/ClickHouse/ClickHouse/blob/master/contrib/snappy-cmake/CMakeLists.txt#L21-L25

so I think the condition should be `if (ARCH_AMD64 AND ENABLE_SSE42)`

Reply:
Yes, I tried it and it works.
```cmake
if (ARCH_AMD64 AND ENABLE_SSSE3)
    set (SNAPPY_HAVE_SSSE3 1)
else ()
    set (SNAPPY_HAVE_SSSE3 0)
endif ()

if (ARCH_AMD64 AND ENABLE_SSE42)
    set (SNAPPY_HAVE_X86_CRC32 1)
else ()
    set (SNAPPY_HAVE_X86_CRC32 0)
endif ()
```
How about it?

Reply:
looks good